.github/workflows/publish-on-pypi.yml

```diff
@@ -22,32 +23,63 @@ jobs:
     - name: Install Python dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install --upgrade setuptools requests
+        pip install --upgrade setuptools requests
+        pip install invoke
+        pip install -e .
 
     - name: Check released tag
       env:
         GITHUB_SHA: ${{ github.sha }}
         GITHUB_REF: ${{ github.ref }}
       run: python .github/workflows/check_released_tag.py
 
-    - name: Install package
-      run: python -m pip install -e .
+    - name: Generate changelog
+      uses: charmixer/auto-changelog-action@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
 
-    - name: Assert package version
-      env:
-        TAG_VERSION: ${{ github.ref }}
-      run: python ./.ci/assert_version.py
+    - name: Update Version
+      run: invoke setver -v ${GITHUB_REF#refs/tags/}
 
-    - name: Build source distribution
-      run: python ./setup.py sdist
+    - name: Generate New OpenAPI JSON
+      run: invoke update-openapijson
 
-    # This tests that publication to PyPI is possible, before properly publishing
-    - name: Publish package to TestPyPI
-      uses: pypa/gh-action-pypi-publish@master
+    - name: Commit files
+      run: |
+        git config --local user.email "dev@optimade.org"
+        git config --local user.name "optimade_devs"
+        mv CHANGELOG.md docs/
+
+        git add docs/CHANGELOG.md
+        git add openapi/*
+        git commit -m 'Release Updates' && echo ::set-env name=push::1 || echo "No changes to CHANGELOG.md"
+
+    - name: Push changes
+      uses: ad-m/github-push-action@master
       with:
-        user: __token__
-        password: ${{ secrets.test_pypi_password }}
-        repository_url: https://test.pypi.org/legacy/
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+
+  publish:
+    runs-on: ubuntu-latest
+    needs:
```

Comment:
A missed needs or an intentional needs?

Reply:
Same as above, trying to constrain a workflow to a certain set of steps.